database/sqlite/sqlite_health.c

```diff
@@ -940,3 +940,271 @@ int alert_hash_and_store_config(
 
     return 1;
 }
+
+void write_alert_analytics(BUFFER *b, char *name, time_t in_clear, uint8_t t_in_clear, uint8_t n_in_clear, time_t in_warn, uint8_t t_in_warn, uint8_t n_in_warn, time_t in_crit, uint8_t t_in_crit, uint8_t n_in_crit, uint8_t no_of_alerts, BUFFER *db_entries) {
+    buffer_sprintf(b, "\t\t\t\t\"%s\": {\n", name);
+    buffer_sprintf(b, "\t\t\t\t\"db_entries\": [ %s\n ],\n", buffer_tostring(db_entries));
+    buffer_sprintf(b, "\t\t\t\t\"num_of_alerts\": %d,\n", no_of_alerts);
+
+    buffer_strcat(b, "\t\t\t\t\t\"CLEAR\": {\n");
+    buffer_sprintf(b, "\t\t\t\t\t\t\"time_spent\": %ld,\n", in_clear);
+    buffer_sprintf(b, "\t\t\t\t\t\t\"times_in_status\": %d,\n", t_in_clear);
+    buffer_sprintf(b, "\t\t\t\t\t\t\"sent_notifications\": %d\n", n_in_clear);
+    buffer_strcat(b, "\t\t\t\t\t},\n");
+
+    buffer_strcat(b, "\t\t\t\t\t\"WARNING\": {\n");
+    buffer_sprintf(b, "\t\t\t\t\t\t\"time_spent\": %ld,\n", in_warn);
+    buffer_sprintf(b, "\t\t\t\t\t\t\"times_in_status\": %d,\n", t_in_warn);
+    buffer_sprintf(b, "\t\t\t\t\t\t\"sent_notifications\": %d\n", n_in_warn);
+    buffer_strcat(b, "\t\t\t\t\t},\n");
+
+    buffer_strcat(b, "\t\t\t\t\t\"CRITICAL\": {\n");
+    buffer_sprintf(b, "\t\t\t\t\t\t\"time_spent\": %ld,\n", in_crit);
+    buffer_sprintf(b, "\t\t\t\t\t\t\"times_in_status\": %d,\n", t_in_crit);
+    buffer_sprintf(b, "\t\t\t\t\t\t\"sent_notifications\": %d\n", n_in_crit);
+    buffer_strcat(b, "\t\t\t\t\t}\n");
+
+    buffer_strcat(b, "\t\t\t\t}");
+}
+
+#define SQL_GET_ALERT_ANALYTICS(guid, start, guid2, start2, end) "SELECT name, chart, new_status, when_key, exec_run_timestamp, alarm_event_id FROM health_log_%s \
+where source like '%%usr/lib/netdata/conf.d/health.d/%%' \
+and when_key < %ld \
+group by name, chart \
+having max(alarm_event_id) \
+UNION ALL \
+SELECT name, chart, new_status, when_key, exec_run_timestamp, alarm_event_id FROM health_log_%s \
+where source like '%%usr/lib/netdata/conf.d/health.d/%%' \
+and (when_key >= %ld and when_key < %ld) \
+order by name, chart, when_key; ", guid, start, guid2, start2, end
+int sql_get_alert_analytics(RRDHOST *host, time_t start, time_t end, BUFFER *b) {
+    sqlite3_stmt *res = NULL;
+    int rc, cnt = 0;
+    char command[MAX_HEALTH_SQL_SIZE + 1];
+
+    BUFFER *db_entries = buffer_create(10000);
+
+    char *name, *chart, *curr_alert = NULL, *curr_chart = NULL;
+    time_t in_clear = 0, in_warn = 0, in_crit = 0, in_other = 0, total_range = 0, range = 0;
+    uint8_t t_in_clear = 0, t_in_warn = 0, t_in_crit = 0, t_in_other = 0;
+    uint8_t n_in_clear = 0, n_in_warn = 0, n_in_crit = 0;
+    uint8_t no_of_alerts = 0, chart_change = 0;
+    time_t when, exec_run_timestamp;
+    int db_entries_c=0;
+    RRDCALC_STATUS status, last_status = RRDCALC_STATUS_UNDEFINED;
```

Comment:
If the status is not CLEAR, WARNING, CRITICAL - let's make it OTHER.

Reply:
Yes, I know now what you meant when we discussed it. We need to leave it like that, it's like a start point for unknown (i.e. not critical, warning, clear) statuses, and I use that because it is a real RRDCALC status and can do comparisons with the others.

It is safe to assume all other statuses will fall under that.